### PR TITLE
docs: Test Workflow docs

### PR DESCRIPTION
**Pull request by tkonieczny**

### CODE REVIEW

File: docs/docs/articles/test-workflows.md [hunk 1]
```diff
@@ -279,32 +279,46 @@ or on the Step level:
...
-      paths:
-      - '**/*'
+  workingDir: /data/repo/test/cypress/executor-tests/cypress-13/cypress/videos
+  artifacts:
+    paths:
+    - '**/*'
+```
+
+### condition: always
+It's a common thing to save artifacts in case of failure. That's when setting `condition: always` may be needed - to execute artifacts saving step even if the previous ones failed.
```
Comment (rangoo94) [inline docs/docs/articles/test-workflows.md:351]: By default the `artifacts` will have `condition: always` for the same level. i.e.

```yaml
- shell: abc
  artifacts:
    paths: ["**/*"]
```

is equivalent of:
```yaml
- steps:
  - shell: abc
  - condition: always
    artifacts:
      paths: ["**/*"]
```

File: docs/docs/articles/test-workflow-templates.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
+# Test Workflow Template
+
+TestWorkflow Templates allow you to define common properties/traits that you can reuse 
+consistently across your TestWorkflows. 
+
+For example, with TestWorkflow Templates you can easily 
+- define specific ENV variables to be passed to all containers
```
Comment (jfermi) [inline docs/docs/articles/test-workflow-templates.md:7]:
```suggestion
- define specific ENV variables to be passed to all containers.
```

File: docs/docs/articles/test-workflow-templates.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+- define a common Git Source for your tests
```
Comment (jfermi) [inline docs/docs/articles/test-workflow-templates.md:8]:
```suggestion
- define a common Git Source for your tests.
```

File: docs/docs/articles/test-workflow-templates.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+- define common setup or tearDown steps for your tests
```
Comment (jfermi) [inline docs/docs/articles/test-workflow-templates.md:9]:
```suggestion
- define common setup or tearDown steps for your tests.
```

File: docs/docs/articles/test-workflow-templates.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+- define specific commands or images to use for your tests, while still allowing a TestWorkflow to set
+  the version of the testing tool
```
Comment (jfermi) [inline docs/docs/articles/test-workflow-templates.md:11]:
```suggestion
  the version of the testing tool.
```

File: docs/docs/articles/test-workflow-templates.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+- etc.
+
+## How does it work?
+
+A TestWorkflowTemplate definition follows the same structure as a TestWorkflow, with the additional
+possibility to define configuration parameters that can be specified when the template is used.
```
Comment (jfermi) [inline docs/docs/articles/test-workflow-templates.md:17]:
```suggestion
ability to define configuration parameters that can be specified when the template is used.
```

File: docs/docs/articles/test-workflow-templates.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+TestWorkflowTemplates are "merged" into a referring TestWorkflow at run-time and the 
+resulting TestWorkflow is executed by Testkube.
+
+For example, the below template defines the basics for running a K6 test, with parameters for which
```
Comment (jfermi) [inline docs/docs/articles/test-workflow-templates.md:22]:
```suggestion
For example, the template below defines the basics for running a k6 test, with parameters for which
```

File: docs/docs/articles/test-workflow-templates.md [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+      type: string
+      default: ""
+  steps: # steps to be executed
+  - name: Run k6 tests
+    container:
+      image: grafana/k6:{{ config.version }} # default values are assigned
+    shell: k6 run {{ config.params }}
+```
+
+The template can then be "used" in a corresponding from TestWorkflow step:
```
Comment (jfermi) [inline docs/docs/articles/test-workflow-templates.md:47]: Is this step called "from"? If it is, I would highlight it (``).

File: docs/docs/articles/test-workflows.md (hunk 1 above)
Comment (tkonieczny) [reply docs/docs/articles/test-workflows.md]: I added different versions with additional descriptions

**Review by jfermi — COMMENTED** [review]
Some editing suggestions.

File: docs/docs/articles/test-workflow-templates.md (hunk 8 above)
Comment (tkonieczny) [reply docs/docs/articles/test-workflow-templates.md]: fixed (typo)

**Review by vsukhin — APPROVED** [review]

**Comment by tkonieczny** [comment]
Docs preview failed here - I checked `docusaurus build` locally, and everything seems fine - merging

**Pull request merged**